src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Node/Generics/JsonValuePrimitiveBooleanConverter.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Nodes.Generics;
+using System.Text.Json.Serialization.Metadata;
+
+namespace System.Text.Json.Serialization.Converters.Node.Generics
+{
+    internal sealed class JsonValuePrimitiveBooleanConverter : JsonValuePrimitiveConverter<bool>
```

Comment:
It's not efficient to create dedicated converter class for each generic instantiation.

Instead, create a converter factory for `Generic<T>`, and invokes the converter of T. Check how `NullableConverter`, `FSharpOptionConverter` and `JsonCollectionConverter` invokes `JsonConverter<TElement>`.

Reply:
I thought the performance of this approach was better

Reply:
While it may improve performance to inline one level of virtual call, it can be easily compensated by other stuff like the ConcurrentDictionary lookup. It's way not maintainable.

Reply:
You are right, and this code already removed